crates/bin/pcli/src/command/init.rs

```diff
@@ -102,6 +99,25 @@ pub enum SoftKmsInitCmd {
     },
 }
 
+// Reusable function for prompting interactively for key material.
+fn prompt_for_password(msg: &str) -> Result<String> {
+    let mut password = String::new();
+    // The `rpassword` crate doesn't support reading from stdin, so we check
+    // for an interactive session. We must support non-interactive use cases,
+    // for integration with other tooling.
+    if std::io::stdin().is_terminal() {
+        password = rpassword::prompt_password(msg)?;
+    } else {
+        while let Ok(n_bytes) = std::io::stdin().lock().read_to_string(&mut password) {
+            if n_bytes == 0 {
+                break;
```

Comment:
I think we want to continue here instead? otherwise the loop doesn't make much sense